examples/src/main/python/sql/streaming/structured_network_wordcount_session_window.py

```diff
@@ -71,55 +72,57 @@
     # Split the lines into words, retaining timestamps, each word become a sessionId
     events = lines.select(
         explode(split(lines.value, " ")).alias("sessionId"),
-        lines.timestamp.cast("long"),
+        lines.timestamp,
     )
 
     # Type of output records.
     session_schema = StructType(
         [
             StructField("sessionId", StringType()),
             StructField("count", LongType()),
-            StructField("start", LongType()),
-            StructField("end", LongType()),
+            StructField("start", TimestampType()),
+            StructField("end", TimestampType()),
         ]
     )
     # Type of group state.
     # Omit the session id in the state since it is available as group key
     session_state_schema = StructType(
         [
             StructField("count", LongType()),
-            StructField("start", LongType()),
-            StructField("end", LongType()),
+            StructField("start", TimestampType()),
+            StructField("end", TimestampType()),
         ]
     )
 
     def func(
-        key: Any, pdf_iter: Iterable[pd.DataFrame], state: GroupState
+        key: Any, pdfs: Iterable[pd.DataFrame], state: GroupState
     ) -> Iterable[pd.DataFrame]:
         if state.hasTimedOut:
             count, start, end = state.get
             state.remove()
+            (session_id,) = key
             yield pd.DataFrame(
                 {
-                    "sessionId": [key[0]],
+                    "sessionId": [session_id],
                     "count": [count],
                     "start": [start],
                     "end": [end],
                 }
             )
         else:
-            start = math.inf
-            end = 0
-            count = 0
+            pdf_iter = iter(pdfs)
```

Comment:
Hm, why should we do this? I think you can directly iterate `pdfs` over.

Also, I think the previous value initialization logic looked making sense. This example now would fail if the function is executed with an empty input.

Reply:
Now start and end are TimestampType, I don't know what should be reasonable min and max values for then

Reply:
I thin you can just saw it as `None`

Reply:
start = min(start, pdf["timestamp"].min())
  end = max(end, pdf["timestamp"].max())
These statement expects a non-empty start and end value.
Also, can this callback ever be invoked with empty records dataframe?

Reply:
Hm, when timeout is reached, it goes to the above branch so technically it won't be. Okay, LGTM but can we avoid `iter(pdfs)` conversion? It's already an iterator.

Reply:
It seems that the pandas udf expects Iterable instead of Iterator as input signature, so I can't change the input type annotation. And this conversion is necessary for us to iterate the first element and then traverse the rest. Technically we know pdfs is an iterator and the conversion is unnecessary.(I tested it locally and it worked without conversion). But the linter mypy will complain.

Reply:
Hm .. this is actually a general issue to fix. Okay, LGTM